### PR TITLE
clear out SDL events to center mouse

### LINKED ISSUE #1594: warping to a map from the command-line always starts me rotated
If I run `woof -warp 1` for example I will start rotated to the right. The amount seems to be based on the mouse sensitivity/acceleration; turning mouse sensitivity to 0 means I'm not rotated.

Any other time I start a map, like through the new game menu or if I hit the "restart map" key, I'm facing the right way.

Using Linux, woof git compiled from source.

### PR DESCRIPTION
**Pull request by rfomin**
Fix #1594 

Taken from Chocolate Doom.

### CODE REVIEW

**Comment by ceski-1** [comment]
This doesn't fix the problem for me unfortunately (Arch, KDE Plasma, SDL 2.30.1-1).

**Comment by rfomin** [comment]
> This doesn't fix the problem for me unfortunately (Arch, KDE Plasma, SDL 2.30.1-1).

How about xfce4? It works for me.

**Comment by rfomin** [comment]
@ceski Right, the previous attempt doesn't work (it's a bit inconsistent on my system).

I cannot replicate the problem with 2786068 so far.

**Comment by ceski-1** [comment]
Okay, I tested https://github.com/fabiangreffrath/woof/pull/1596/commits/27860684d350f8874fb8fefa06f74aa7863237ba.

xfce4 seems fine for all modes (windowed, fullscreen, exclusive fullscreen) and it always captures the mouse correctly now.

For KDE Plasma, exclusive fullscreen seems fine. However, for windowed or fullscreen, the mouse isn't captured on most launches ("I" is still shown until moving mouse). If the mouse happens to be in the lower right part of the screen when the game is launched, and I very slightly move it when the wipe is almost finished (e.g. to prep for a strafe run), the view will suddenly snap rotate way too much (90-180 degrees). So it seems like for this desktop environment, SDL isn't grabbing the mouse correctly.

File: src/i_video.c [hunk 1]
```diff
@@ -1729,6 +1729,12 @@ void I_InitGraphics(void)
     ResetResolution(CurrentResolutionHeight(), true);
     CreateSurfaces(video.pitch, video.height);
     ResetLogicalSize();
+
+    // clear out any events waiting at the start and center the mouse
+    SDL_Event dummy;
+    while (SDL_PollEvent(&dummy))
+        ;
+    I_ResetRelativeMouseState();
```
Comment (ceski-1) [inline src/i_video.c:1737]: This seems safer so other important events aren't discarded:

```c
    SDL_PumpEvents();
    SDL_FlushEvent(SDL_MOUSEMOTION);
    I_ResetRelativeMouseState();
```

**Comment by rfomin** [comment]
I guess this fix is not worth it. Should I close it?

File: src/i_video.c (hunk 1 above)
Comment (rfomin) [reply src/i_video.c:1737]: But it still doesn't help with KDE Plasma?

Comment (ceski-1) [reply src/i_video.c:1737]: No it's the same behavior but we aren't discarding events like "gamepad connected".

**Comment by ceski-1** [comment]
> I guess this fix is not worth it. Should I close it?

It seems to fix xfce4 though and seems like good practice.

File: src/i_video.c (hunk 1 above)
Comment (ceski-1) [reply src/i_video.c:1737]: > But it still doesn't help with KDE Plasma?

And to be clear, the fix still helps the original issue so it no longer instantly rotates view when the mouse isn't even moved.

Comment (rfomin) [reply src/i_video.c]: > This seems safer so other important events aren't discarded:

Done in https://github.com/fabiangreffrath/woof/pull/1596/commits/1d6d3d59e06bd6b59b16aa3fdf8342a87a1aa451

**Comment by fabiangreffrath** [comment]
I still don't get the order of calls. Why not center the mouse first and then eat all events that this triggers?

**Comment by rfomin** [comment]
> I still don't get the order of calls. Why not center the mouse first and then eat all events that this triggers?

Do you mean this?
```c
    I_ResetRelativeMouseState();
    SDL_PumpEvents();
    SDL_FlushEvent(SDL_MOUSEMOTION);
```

Then it won't work for me. I guess we don't care about `I_ResetRelativeMouseState` events, there are some SDL events before video initialisation. I don't know, this fix is hacky and the behaviour may change in the next SDL/WM Manager releases. @fabiangreffrath Is this problem present on your system?

**Comment by fabiangreffrath** [comment]
> @fabiangreffrath Is this problem present on your system?

Yes, I also experienced this occasionally on my GNOME desktop. But I never bothered enough to find a pattern when it occurred. 

I'd guess it's rather related to X11 vs. Wayland, than to the actual desktop environment running on top.

**Comment by fabiangreffrath** [comment]
The bug happens to me in windowed and fullscreen mode (didn't check exclusive fullscreen). This PR fixes it!

**Review by fabiangreffrath — APPROVED** [review]

**Comment by kitchen-ace** [comment]
This does seem to fix it for me as well. Thanks! I guess @ceski-1 is also experiencing something else as well.

**Comment by jackrjli** [comment]
I had this issue too (Fedora with xfce), and this fixed it for me. :+1:

**Pull request merged**